### PR TITLE
Update sorting logic

### LINKED ISSUE #427: Allow artifact page filtering by number of substats
The option to:

- Show everything but 4-line artifacts
- Show only 4-line artifacts

### LINKED ISSUE #463: Favorite characters are sorted above NEW characters in the add character modal
https://discord.com/channels/785153694478893126/787340960173391903/990346881815175268

NEW characters should always be on top

### LINKED ISSUE #625: Maintain same sorting settings on character popup to character page
Currently the weapon and element filtering carries over between the two. But the sorting doesnt

Claymore Cryo (and Hydro) by Name:
image

Claymore Cryo by Level:
image

### LINKED ISSUE #628: Add filter based on RV
So you can show/hide artifacts under or over a certain RV% based on your filters

For example: only show artifacts that are currently under 300% RV
or: only show artifacts that have the potential to be over 700% RV

That way you can then use the unequip/exclude/lock/delete functions to manage inventory based on RV as well

### PR DESCRIPTION
**Pull request by frzyc**
- Resolve #463 - Update sorting logic
- Resolve #427 - Add artifact filter for number of substats
- Resolve #628 - Add artifact filter for RV
- Fix #625 - Sync sorting state

### CODE REVIEW

**Comment by nguyentvan7** [comment]
image
Ascending sort puts favorite characters last

File: src/Util/CharacterSort.ts [hunk 1]
```diff
@@ -2,42 +2,53 @@ import { AllCharacterSheets } from "../Data/Characters/CharacterSheet";
...
 export function characterSortConfigs(database: ArtCharDatabase, characterSheets: AllCharacterSheets): SortConfigs<CharacterSortKey, CharacterKey> {
   return {
-    new: {
-      getValue: (ck) => database.chars.get(ck as CharacterKey) ? 0 : 1,
-      tieBreaker: "name"
+    new: (ck) => database.chars.get(ck as CharacterKey) ? 0 : 1,
+    name: (ck) => i18n.t(`charNames_gen:${charKeyToCharName(ck, database.gender)}`).toString(),
+    level: (ck) => {
+      const char = database.chars.get(ck as CharacterKey)
+      return char ? char.level * char.ascension : 0
```
Comment (nguyentvan7) [inline src/Util/CharacterSort.ts:16]: All Ascension 0 characters will return with a level of 0

File: src/Util/WeaponSort.ts [hunk 2]
```diff
@@ -2,26 +2,25 @@ import { AllWeaponSheets } from "../Data/Weapons/WeaponSheet";
...
   return {
-    level: {
-      getValue: wp => wp.level * wp.ascension ?? 0,
-      tieBreaker: "rarity"
-    },
-    rarity: {
-      getValue: wp => weaponSheets(wp.key).rarity,
-      tieBreaker: "level"
-    }
+    level: wp => wp.level * wp.ascension ?? 0,
```
Comment (nguyentvan7) [inline src/Util/WeaponSort.ts:9]: Same as character, all ascension 0 weapons will have level 0 returned

**Comment by frzyc** [comment]
> image
> 
> Ascending sort puts favorite characters last, I assume they should still be coming first?

You are right. added a way for "new"/"favorite" to bypass asending/desending

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**